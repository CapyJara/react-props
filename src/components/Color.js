import React from 'react';
import PropTypes from 'prop-types';

// eslint-disable-next-line react/prop-types
export default function Color({ name, rgb, hex }) {
  if(!hex) {
    hex = Number(`${rgb.red}${rgb.green}${rgb.blue}`).toString(16);
    if(hex.length < 2) {
      hex = '0' + hex;
    }
  }
  
  return (
    <dl>
      { name ? 
      <> <dt>Name</dt> <dd>{name}</dd> </> 
        : <> <dt>Name</dt> <dd>{hex}</dd> </>}

      <dt>Red</dt>
      <dd>{rgb.red}</dd>

      <dt>Green</dt>
      <dd>{rgb.green}</dd>

      <dt>Blue</dt>
      <dd>{rgb.blue}</dd>

      { hex && <> <dt>Hex</dt>
      <dd>{hex}</dd> </>}

    </dl>
  );
}

Color.propTypes = {
  name: PropTypes.string,
  rgb: PropTypes.object.isRequired,
  hex: PropTypes.string,
};
