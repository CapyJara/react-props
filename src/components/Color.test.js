import React from 'react';
import { shallow } from 'enzyme';
import Color from './Color';

describe('Dog component', () => {
  it('renders a pink with all fields', () => {
    const wrapper = shallow(<Color name="Pink" rgb={ { red: 255, green: 105, blue: 180 } } hex="ff69b4" />);
    expect(wrapper).toMatchSnapshot();
  });
  it('renders a pink without hex', () => {
    const wrapper = shallow(<Color name="Pink" rgb={ { red: 255, green: 105, blue: 180 } } />);
    expect(wrapper).toMatchSnapshot();
  });
  it('renders a pink without name or hex', () => {
    const wrapper = shallow(<Color rgb={ { red: 255, green: 105, blue: 180 } } />);
    expect(wrapper).toMatchSnapshot();
  });
});
