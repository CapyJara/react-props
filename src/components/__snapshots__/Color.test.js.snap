// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dog component renders a pink with all fields 1`] = `
<dl>
   
  <dt>
    Name
  </dt>
   
  <dd>
    Pink
  </dd>
   
  <dt>
    Red
  </dt>
  <dd>
    255
  </dd>
  <dt>
    Green
  </dt>
  <dd>
    105
  </dd>
  <dt>
    Blue
  </dt>
  <dd>
    180
  </dd>
   
  <dt>
    Hex
  </dt>
  <dd>
    ff69b4
  </dd>
   
</dl>
`;

exports[`Dog component renders a pink without hex 1`] = `
<dl>
   
  <dt>
    Name
  </dt>
   
  <dd>
    Pink
  </dd>
   
  <dt>
    Red
  </dt>
  <dd>
    255
  </dd>
  <dt>
    Green
  </dt>
  <dd>
    105
  </dd>
  <dt>
    Blue
  </dt>
  <dd>
    180
  </dd>
   
  <dt>
    Hex
  </dt>
  <dd>
    f34989c
  </dd>
   
</dl>
`;

exports[`Dog component renders a pink without name or hex 1`] = `
<dl>
   
  <dt>
    Name
  </dt>
   
  <dd>
    f34989c
  </dd>
   
  <dt>
    Red
  </dt>
  <dd>
    255
  </dd>
  <dt>
    Green
  </dt>
  <dd>
    105
  </dd>
  <dt>
    Blue
  </dt>
  <dd>
    180
  </dd>
   
  <dt>
    Hex
  </dt>
  <dd>
    f34989c
  </dd>
   
</dl>
`;
